### ISSUE TITLE
NUnit1001 - false positive for arguments with nullable enum parameters

### ISSUE THREAD

**Issue by maksim-sovkov** [nunit/nunit.analyzers#939]
```cs
public class Tests
{
    public enum TestEnum { A, B, C }

    [TestCase(null)]
    public void Test1(TestEnum? e)
    {
        Assert.Pass();
    }

    [TestCase(TestEnum.A)]
    public void Test2(TestEnum? e)
    {
        Assert.Pass();
    }
}
```

```
error NUnit1001: The value of the argument at position '0' of type <null> cannot be assigned to the parameter 'e' of type TestProject1.Tests.TestEnum? (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit1001.md)
error NUnit1001: The value of the argument at position '0' of type TestProject1.Tests.TestEnum cannot be assigned to the parameter 'e' of type TestProject1.Tests.TestEnum? (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit1001.md)
```

**Comment by NinoFloris** [comment]
We have a similar issue for normal enums (in arrays) as well https://github.com/npgsql/npgsql/actions/runs/18922079735/job/54020576719?pr=6276 the errors there originate from:

https://github.com/npgsql/npgsql/blob/4f7cf71ff1b8d8057eac2e438d6d5401fabeeb22/test/Npgsql.Tests/MultipleHostsTests.cs#L218-L222

**Comment by maksim-sovkov** [comment]
> We have a similar issue for normal enums (in arrays) as well https://github.com/npgsql/npgsql/actions/runs/18922079735/job/54020576719?pr=6276 the errors there originate from:
> 
> https://github.com/npgsql/npgsql/blob/4f7cf71ff1b8d8057eac2e438d6d5401fabeeb22/test/Npgsql.Tests/MultipleHostsTests.cs#L218-L222

Sounds like a good test case!
I'll add it to my pull request.

**Comment by mikkelbu** [comment]
Thanks for the report and accompanying fix @maksim-sovkov.
